tests/test_external_search.py

```diff
@@ -2,10 +2,488 @@
+
+        fiction_only = self._work(fiction=True)
+        fiction_only.set_presentation_ready()
+        fiction_only.update_external_index(self.search)
+
+        time.sleep(1)
+
+        results = self.search.query_works("age 7 fiction", None, None, None, None, None, None, None)
+        hits = results["hits"]["hits"]
+        eq_(1, len(hits))
+        eq_(unicode(age_range_and_fiction.id), hits[0]['_id'])
+
+    def test_query_works_matches_genre_and_title(self):
+        if not self.search:
+            return
+
+        genre_title = self._work(genre="Biography", title="Abraham Lincoln")
+        genre_title.set_presentation_ready()
+        genre_title.update_external_index(self.search)
+
+        genre_only = self._work(genre="Biography")
+        genre_only.set_presentation_ready()
+        genre_only.update_external_index(self.search)
+
+        title_only = self._work(title="Abraham Lincoln")
+        title_only.set_presentation_ready()
+        title_only.update_external_index(self.search)
+
+        time.sleep(1)
+
+        results = self.search.query_works("lincoln biography", None, None, None, None, None, None, None)
+        hits = results["hits"]["hits"]
+        eq_(2, len(hits))
+        eq_(unicode(genre_title.id), hits[0]['_id'])
+        eq_(unicode(title_only.id), hits[1]['_id'])
+
+    def test_query_works_matches_audience_and_author(self):
+        if not self.search:
+            return
+
+        audience_author = self._work(audience=Classifier.AUDIENCE_YOUNG_ADULT, authors="Sherman Alexie")
+        audience_author.set_presentation_ready()
+        audience_author.update_external_index(self.search)
+
+        audience_only = self._work(audience=Classifier.AUDIENCE_YOUNG_ADULT)
+        audience_only.set_presentation_ready()
+        audience_only.update_external_index(self.search)
+
+        author_only = self._work(audience=Classifier.AUDIENCE_ADULT, authors="Sherman Alexie")
+        author_only.set_presentation_ready()
+        author_only.update_external_index(self.search)
+
+        time.sleep(1)
+
+        results = self.search.query_works("sherman alexie ya", None, None, None, None, None, None, None)
+        hits = results["hits"]["hits"]
+        eq_(2, len(hits))
+        eq_(unicode(audience_author.id), hits[0]['_id'])
+        eq_(unicode(author_only.id), hits[1]['_id'])
+
+    def test_query_works_matches_fiction_and_summary(self):
+        if not self.search:
+            return
+
+        fiction_summary = self._work(fiction=True)
+        fiction_summary.summary_text = "A book about a dog"
+        fiction_summary.set_presentation_ready()
+        fiction_summary.update_external_index(self.search)
+
+        nonfiction_summary = self._work(fiction=False)
+        nonfiction_summary.summary_text = "A book about a real dog"
+        nonfiction_summary.set_presentation_ready()
+        nonfiction_summary.update_external_index(self.search)
+
+        time.sleep(1)
```

Comment:
Is there any way to reduce the amount of time we have to wait for the index to get updated? There are a lot of these tests and they all sleep for a second.

Or maybe we could create the entire index in the module setup so that we only have to sleep once? I'm not sure what's the best way to handle this.